feat(web): Add tooltip to production source legends in BreakdownChart

silkeholmebonnen opened the pull request:
## Issue

<!-- If you want to close an issue automatically when your PR is merged, write "Closes X" where X is the PR number. For example: Closes #000 -->
[AVO-234](https://linear.app/electricitymaps/issue/AVO-234/origin-of-electricity-add-legend-tooltip-for-sources)

## Description

<!-- Explains the goal of this PR -->
This PR adds a tooltip to the production source legend list in the BreakdownChart.

### Preview
<img width="418" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/95029552/6ef12ca2-6e2c-4993-9c38-83db6054dc60">
<img width="419" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/95029552/e75c84e3-73e0-4114-9b9e-c0ab60ee51fc">

Mobile:
<img width="373" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/95029552/702e0d81-cd2d-4b9e-889a-c56e5cd1f5b0">
<img width="371" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/95029552/9af40eec-6978-4bd6-a04b-7998e31a7d3e">

<!-- Please add screenshots and/or gif that shows visual changes (if applicable) -->

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

silkeholmebonnen commented:
@Alportan Do you have any feedback?

Alportan commented:
> @Alportan Do you have any feedback?

Looking good! 😎 Maybe one small thing, could we use the frosted background blur on the modal? 🙌 
<img width="671" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/149581095/b1a97ee1-dc22-4c5f-a3a1-c318a5ad3061">

VIKTORVAV99 commented:
> > @Alportan Do you have any feedback?
> 
> Looking good! 😎 Maybe one small thing, could we use the frosted background blur on the modal? 🙌 <img alt="image" width="671" src="https://private-user-images.githubusercontent.com/149581095/347006743-b1a97ee1-dc22-4c5f-a3a1-c318a5ad3061.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1Mzc5MDAsIm5iZiI6MTcyMDUzNzYwMCwicGF0aCI6Ii8xNDk1ODEwOTUvMzQ3MDA2NzQzLWIxYTk3ZWUxLWRjMjItNGM1Zi1hM2ExLWMzMThhNWFkMzA2MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwOVQxNTA2NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNjQwMDYwYjk5MTkxYjQ0MzIzYmZmMDI5NmFjZDM4MTM2OTkzMmMxNDU0YzZiNjkyZThlOWVjMGU3Y2FjMjEwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4-DlaxebFqNIAoN5Yw8EaywGKrM_8LGbersp-Yjyet0">
> 
> I believe we already use it on a few components/elements.

I have been using background-blur (not sm) to not get the edges to bleed tough, it would be nice if it was the same here :)

silkeholmebonnen commented:
I added the blur and I also changed the mobile version to not be in the middle of the screen, but right above or below the production source icons, because thats what it looked like in figma. Will you take a look at it again @Alportan, and let me know if you would rather have in be in the center on mobile.

Alportan commented:
> I added the blur and I also changed the mobile version to not be in the middle of the screen, but right above or below the production source icons, because thats what it looked like in figma. Will you take a look at it again @Alportan, and let me know if you would rather have in be in the center on mobile.

Looks good with the blur! 🫶

Regarding the center modal on mobile, that's a good question. I'm unsure about it for now because adding a tooltip on top of the graph might be disruptive, as the main goal is to help users identify sources on the graph.

It's important to keep the design consistent and avoid too many custom components or logic. However, we should review all current use-cases first. There’s a future task to [assess and unify tooltips](https://www.notion.so/electricitymaps/Revise-and-unify-all-tooltips-d4f78af5dcba4ab695b940f33f0ed4a0?pvs=4), so we can start with this implementation and refine it later as needed.

silkeholmebonnen commented:
> > I added the blur and I also changed the mobile version to not be in the middle of the screen, but right above or below the production source icons, because thats what it looked like in figma. Will you take a look at it again @Alportan, and let me know if you would rather have in be in the center on mobile.
> 
> Looks good with the blur! 🫶
> 
> Regarding the center modal on mobile, that's a good question. I'm unsure about it for now because adding a tooltip on top of the graph might be disruptive, as the main goal is to help users identify sources on the graph.
> 
> It's important to keep the design consistent and avoid too many custom components or logic. However, we should review all current use-cases first. There’s a future task to [assess and unify tooltips](https://www.notion.so/electricitymaps/Revise-and-unify-all-tooltips-d4f78af5dcba4ab695b940f33f0ed4a0?pvs=4), so we can start with this implementation and refine it later as needed.

Okay, I'll keep it as it is for now then:)

silkeholmebonnen commented:
/review

VIKTORVAV99 commented on web/src/features/charts/ProductionSourceLegendList.tsx:
```diff
@@ -10,10 +13,59 @@ export default function ProductionSourceLegendList({
+      isMobile={isMobile}
+    >
+      <div className={twMerge('flex w-fit flex-row gap-1.5 py-1', className)}>
+        {sources.map((source, index) => (
+          <ProductionSourceLegend key={index} electricityType={source} />
+        ))}
```
As the bot suggested you can use the source object directly as a key here as well just as you did later in the code.

VIKTORVAV99 reviewed (approved):
LGTM! 🎉

Just one minor thing but nothing blocking! Nice work!
